PathPlanning/DStar/dstar.py

```diff
@@ -116,7 +117,7 @@ def process_state(self):
                     self.insert(y, x.h + x.cost(y))
                 else:
                     if y.parent != x and y.h > x.h + x.cost(y):
-                        self.insert(y, x.h)
+                        self.insert(x, x.h)
```

Comment:
Was it a bug?

Reply:
Please see https://www.ri.cmu.edu/pub_files/pub3/stentz_anthony__tony__1994_2/stentz_anthony__tony__1994_2.pdf, page 11. Both changes are necessary, otherwise the program will stuck when new obstacles are added. 

Also check out https://www.cs.cmu.edu/~motionplanning/lecture/AppH-astar-dstar_howie.pdf, page 51, I think the instructions on this page indicate the need to change elif to if.

So is this a correct modification？